<?xml version="1.0" ?>
<launch>
  <arg name="vehicle" default="$(env ROS_HOSTNAME)" />
  <arg name="ns" default="/$(arg vehicle)" />
  <!-- Type of node.  Current options are 'robot' and 'base' -->
  <arg name="type" default="robot" />
  <!-- Merger type - 0: Octomap, 1: PCL-Kyle, 2: PCL-Jessup -->
  <arg name="merger" default="0" />
  <!-- Octomap type - 0: Binary, 1: Full -->
  <arg name="octoType" default="0" />
  <!-- Map resolution.  If different res needed for each need to change code -->
  <arg name="resolution" default="0.2" />
  <!-- Size of map differences to trigger a merge -->
  <arg name="mapThresh" default="100" />
  <!-- Topics to subscribe and publish -->
  <arg name="mapTopic" default="octomap_binary" />
  <arg name="neighborsTopic" default="neighbor_maps" />
  <arg name="mergedTopic" default="merged_map" />

  <node ns="$(arg ns)" name="octomap_merger" pkg="octomap_merger" type="octomap_merger_node" output="screen">
    <param name="type" value="$(arg type)" />
    <param name="merger" value="$(arg merger)" />
    <param name="octoType" value="$(arg octoType)" />
    <param name="resolution" value="$(arg resolution)" />
    <param name="mapThresh" value="$(arg mapThresh)" />
    <param name="mapTopic" value="$(arg mapTopic)" />
    <param name="neighborsTopic" value="$(arg neighborsTopic)" />
    <param name="mergedTopic" value="$(arg mergedTopic)" />
  </node>
</launch>
